hw/top_earlgrey/data/pins_cw310.xdc

```diff
@@ -78,8 +78,19 @@ set_property -dict { PACKAGE_PIN AF14  IOSTANDARD LVCMOS18 } [get_ports { IO_UPH
 set_property -dict { PACKAGE_PIN P18   IOSTANDARD LVCMOS33 } [get_ports { IO_UPHY_SENSE }]; #USRUSB_VBUS_DETECT
 set_property -dict { PACKAGE_PIN AE15  IOSTANDARD LVCMOS18 } [get_ports { IO_UPHY_OE_N }]; #USRUSB_OE
 set_property -dict { PACKAGE_PIN V17   IOSTANDARD LVCMOS18 } [get_ports { IO_UPHY_D_RX }]; #USRUSB_RCV
-#set_property -dict { PACKAGE_PIN AE16  IOSTANDARD LVCMOS18 } [get_ports { IO_UPHY_SPD  }]; #USRUSB_SPD
-#set_property -dict { PACKAGE_PIN AF15  IOSTANDARD LVCMOS18 } [get_ports { IO_UPHY_SUS  }]; #USRUSB_SUS
+set_property -dict { PACKAGE_PIN AE16  IOSTANDARD LVCMOS18 } [get_ports { IO_UPHY_SPD  }]; #USRUSB_SPD
+set_property -dict { PACKAGE_PIN AF15  IOSTANDARD LVCMOS18 } [get_ports { IO_UPHY_SUS  }]; #USRUSB_SUS
+
+## USB input delay to accommodate T_FST and the PHY's sampling logic
+## The PHY expects to only see up to one transient / fake SE0. The phase
+## relationship with the PHY's sampling clock is arbitrary, but for simplicity,
```

Comment:
I am a bit confused here. Does the PHY use a sampling clock? I thought it's not clocked. At least in the data sheet I couldn't find any information on this.

Reply:
This is probably a difference in how we use terminology.

I think of the output of a PHY as being "raw" bits, so our PHY includes both the discrete transceiver (the asynchronous TUSB1106) and the sampling logic in the `usb_fs_nb_pe` module. These constraints are satisfying part of the Electrical section of the spec, which specifically identifies itself as the physical layer specification.

Reply:
Ah thanks for clarifying. This makes sense.